integration-cli/docker_cli_exec_test.go

```diff
@@ -232,18 +197,18 @@ func (s *DockerSuite) TestExecParseError(c *check.C) {
 
 	// Test normal (non-detached) case first
 	cmd := exec.Command(dockerBinary, "exec", "top")
-	if _, stderr, code, err := runCommandWithStdoutStderr(cmd); err == nil || !strings.Contains(stderr, "See '"+dockerBinary+" exec --help'") || code == 0 {
-		c.Fatalf("Should have thrown error & point to help: %s", stderr)
-	}
+	_, stderr, code, err := runCommandWithStdoutStderr(cmd)
+	c.Assert(err, checker.NotNil)
+	c.Assert(stderr, checker.Contains, "See '"+dockerBinary+" exec --help'")
+	c.Assert(code, checker.Not(checker.Equals), 0)
```

Comment:
Do we really care about the `exitCode` in this test ? We know for a fact it won't be 0 because it had an error (or if it hadn't, previous test failed so this line won't be called). It's a nit so I'm fine if we keep it too :stuck_out_tongue_closed_eyes:.